client/my-sites/upgrades/checkout-thank-you/header/index.jsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const CheckoutThankYouHeader = React.createClass( {
+	propTypes: {
+		isDataLoaded: React.PropTypes.bool.isRequired,
+		isFreeTrial: React.PropTypes.bool.isRequired,
+		productName: React.PropTypes.string.isRequired
+	},
+
+	renderHeading() {
+		let heading = this.translate( 'Thank you for your purchase!' );
+
+		if ( ! this.props.isDataLoaded ) {
+			heading = this.translate( 'Loading…' );
+		} else if ( this.props.isFreeTrial ) {
+			heading = this.translate( 'Your 14 day free trial starts today!' );
+		}
+
+		return (
+			<h1 className="checkout-thank-you-header__heading">
+				{ heading }
+			</h1>
+		);
+	},
+
+	renderText: function() {
+		let text = this.translate( "You will receive an email confirmation shortly. What's next?" );
+
+		if ( ! this.props.isDataLoaded ) {
```

Comment:
It might be good to refactor this logic away, so you could have a `renderHeadingTwo` function and a `getText` function which it calls. That would mean you can return early in `getText`, which would make it more readable.